### PR TITLE
Feature fix linter config test

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: lib/githooks/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ls -alF /my/project/to/be/linted/.git/hooks/pre-commit
+```
+
+## Debug output toggle
+
+To enable debug logging, change the docker environment variable, in the [pre-commit](./pre-commit) script to "true".
+
+### Versent internal link
+
+See https://versent.atlassian.net/wiki/spaces/TCIP/pages/585203752/Linting
```
Comment (17twenty) [inline lib/githooks/README.md:71]: I'd remove this

File: lib/linters/cfn-lint.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  local RC=0
+  local FILENAME="${1}"
+
+  if echo "${FILENAME}" | grep -e \.yml$ -e \.yaml$ -e \.yaml$ -e \.cfn$ -e \.template$ > /dev/null
+  then
+
+    if head "${FILENAME}"  | grep "AWSTemplateFormatVersion" > /dev/null
+    then
+      log "Invoking cfn-lint on [${FILENAME}]"
+      cfn-lint "${FILENAME}" -i E2541 E2540 W1020 W3002 2>&1
```
Comment (balloonpopper) [inline lib/linters/cfn-lint.sh:25]: Should call out in a comment why these exclusions are in place

File: lib/linters/cfn-lint.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -ou pipefail
+
+declare COMMON_LIB_PATH=""
+COMMON_LIB_PATH="$(dirname "$(dirname "${BASH_SOURCE[0]}")")/common.sh"
+
+#shellcheck disable=SC1090
```
Comment (balloonpopper) [inline lib/linters/cfn-lint.sh:8]: Should call out in a comment why these exclusions are in place

File: lib/linters/pylint.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -ou pipefail
+
+declare COMMON_LIB_PATH=""
+COMMON_LIB_PATH="$(dirname "$(dirname "${BASH_SOURCE[0]}")")/common.sh"
+
+#shellcheck disable=SC1090
```
Comment (balloonpopper) [inline lib/linters/pylint.sh:8]: Should call out in a comment why these exclusions are in place

File: lib/linters/jsonlint.sh [hunk 5]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -e
+
+set -ou pipefail
+
+declare COMMON_LIB_PATH=""
+COMMON_LIB_PATH="$(dirname "$(dirname "${BASH_SOURCE[0]}")")/common.sh"
+
+#shellcheck disable=SC1090
```
Comment (balloonpopper) [inline lib/linters/jsonlint.sh:8]: Should call out in a comment why these exclusions are in place

File: lib/linters/cloudformation-validate.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -ou pipefail
+
+declare COMMON_LIB_PATH=""
+COMMON_LIB_PATH="$(dirname "$(dirname "${BASH_SOURCE[0]}")")/common.sh"
+
+#shellcheck disable=SC1090
```
Comment (balloonpopper) [inline lib/linters/cloudformation-validate.sh:8]: Should call out in a comment why these exclusions are in place

File: lib/linters/shellcheck.sh [hunk 7]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -ou pipefail
+
+declare COMMON_LIB_PATH=""
+COMMON_LIB_PATH="$(dirname "$(dirname "${BASH_SOURCE[0]}")")/common.sh"
+
+# TODO - https://github.com/koalaman/shellcheck/wiki/Ignore
+
+#shellcheck disable=SC1090
```
Comment (balloonpopper) [inline lib/linters/shellcheck.sh:10]: Should call out in a comment why these exclusions are in place

File: lib/linters/yamllint.sh [hunk 8]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -ou pipefail
+
+declare COMMON_LIB_PATH=""
+COMMON_LIB_PATH="$(dirname "$(dirname "${BASH_SOURCE[0]}")")/common.sh"
+
+#shellcheck disable=SC1090
```
Comment (balloonpopper) [inline lib/linters/yamllint.sh:8]: Should call out in a comment why these exclusions are in place

File: lib/linters/yamllint.sh [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+WORKING_DIR="${2}"
+
+# TODO - https://yamllint.readthedocs.io/en/stable/configuration.html
+
+handle()
+{
+  local RC=0
+  local FILENAME="${1}"
+  local USERS_YAML_CONFIG_PATH="${WORKING_DIR}/.yamllintrc"
```
Comment (balloonpopper) [inline lib/linters/yamllint.sh:19]: yamllint's exclusion file is .yamllint not .yamllintrc
https://yamllint.readthedocs.io/en/stable/configuration.html

File: lib/linters/yamllint.sh [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  local DEFAULT_YAML_CONFIG_PATH="./lintball-yamllintrc"
```
Comment (balloonpopper) [inline lib/linters/yamllint.sh:20]: Might want to make this lintball-yamllint to be consistent with the above requested change from .yamllintrc to .yamllint

File: lintball.sh [hunk 11]
```diff
@@ -1,114 +1,130 @@
...
-  echo "Error - filename not passed to the script"
-  usage
+declare RESULT_PREFIX="Lintball result :"
+declare LINTIGNORE_FILENAME=".lintignore"
+
+# Ensure we run the script from the app directory, not the mounted directory
+cwd=$(dirname "${0}")
+cd "${cwd}"
+
+#shellcheck disable=SC1091
```
Comment (balloonpopper) [inline lintball.sh:19]: Should call out in a comment why these exclusions are in place

File: lintball.sh [hunk 12]
```diff
@@ -1,114 +1,130 @@
...
+
+# If the user passes in a git address, clone the repo and lint the changed files
+# (else, the filenames are expected as args)
+if [ -n "${GIT_URI:-""}" ]; then
+  WORKING_DIR="${WORKING_DIR}/git"
+  FILENAMES="TODO"
+  git clone "${GIT_URI}" "${WORKING_DIR}"
+
+  # Checkout the branch we are interested in (in a subshell, so as not to affect PWD)
+  #shellcheck disable=SC2091
```
Comment (balloonpopper) [inline lintball.sh:38]: Should call out in a comment why these exclusions are in place

File: lintball.sh [hunk 13]
```diff
@@ -1,114 +1,130 @@
...
+        set +e
+        ${linter} "${FILENAME}" "${WORKING_DIR}" || RC=$?
+        set -e
+      done
+    else
+      log "DELETED FILE ${FILENAME} - IGNORING"
     fi
-fi
+
+  elsenice
```
Comment (balloonpopper) [inline lintball.sh:112]: elsenice?

File: lib/linters/cfn-lint.sh [hunk 14]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (balloonpopper) [inline lib/linters/cfn-lint.sh:1]: Can this script look for .cfnlintrc as per https://github.com/awslabs/cfn-python-lint?

File: lib/linters/pylint.sh [hunk 15]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
```
Comment (balloonpopper) [inline lib/linters/pylint.sh:1]: Can this script be modified to source pylintrc files?

File: lintball.sh [hunk 16]
```diff
@@ -1,114 +1,130 @@
...
+  then
+    debug "$(ls -alF "${FILENAME}")"
+  fi
+
+  # Does the given .lintignore file exist
+  if [ -f "${LINTIGNORE_PATH}" ] ; then
+
+    debug "Found ${LINTIGNORE_FILENAME} file"
+    # Ignore if file is in lintignore
+    if grep -q "${FILE}" "${LINTIGNORE_PATH}"; then
```
Comment (balloonpopper) [inline lintball.sh:79]: I'd suggest that this either needs to be "^${FILE}$" or we need to call out that the files in the lintignore file are effectively a regex

**Pull request closed without merging**